pennylane/operation.py

```diff
@@ -575,6 +585,28 @@ def decompose(self):
             return self.decomposition(wires=self.wires)
         return self.decomposition(*self.parameters, wires=self.wires)
 
+    @staticmethod
+    def compute_diagonalizing_gates(*params, wires, **hyperparams):
+        r"""Defines a partial representation of this operator as
+        an eigendecompisition.
+
+        Multiplied together, the diagonalizing gates
+        form the unitary :math:`U` in `O = U \Sigma U^{\dagger}`, where
+        :math:`\Sigma` is a diagonal matrix that contains the eigenvalues of :math:`O`.
+
+        This is the static version of ``diagonalizing_gates``, which can be called
+        on the class.
+
+        *Example:*
+
+        >>> qml.PauliX.compute_diagonalizing_gates(wires="q1")
+        [Hadamard(wires=["q1"])]
+
+        Returns:
+            list(qml.Operator): A list of operators.
+        """
+        return None
+
     # pylint:disable=no-self-use
```

Comment:
```suggestion
```

Uses `self` now, so can get rid of this.

Reply:
Good catch!